core/src/views/radio_buttons.rs

```diff
@@ -0,0 +1,118 @@
+}
+
+const ICON_CHECK: &str = "\u{00B7}";
+
+impl RadioButtons {
+    pub fn with_icons<F>(
+        cx: &mut Context,
+        idx: usize,
+        idx_count: usize,
+        builder: F,
+        icon_checked: String,
+        icon_unchecked: String,
+    ) -> Handle<Self>
+    where
+        F: 'static + Fn(&mut Context, usize) + Clone
+    {
+        Self {
+            on_changed: None,
+        }
+            .build2(cx, move |cx| {
+                for idx_available in 0..idx_count {
+                    let checked = idx_available == idx;
+                    let text = if checked { icon_checked.clone() } else { icon_unchecked.clone() };
+                    let one_builder = builder.clone();
+                    RadioButton::new(cx, idx_available, text, move |cx| {
+                        (one_builder)(cx, idx_available);
+                    });
+                }
+            })
+    }
+
+    pub fn new<F>(
+        cx: &mut Context,
+        idx: usize,
+        idx_count: usize,
+        builder: F
+    ) -> Handle<Self>
+        where
+            F: 'static + Fn(&mut Context, usize) + Clone
+    {
+        Self::with_icons(cx, idx, idx_count, builder, ICON_CHECK.to_owned(), "".to_owned())
+    }
+}
+
+impl Handle<RadioButtons> {
+    pub fn on_changed<F>(self, cx: &mut Context, callback: F) -> Self
+        where
+            F: 'static + Fn(&mut Context, usize),
+    {
+        if let Some(view) = cx.views.get_mut(&self.entity) {
+            if let Some(view) = view.downcast_mut::<RadioButtons>() {
+                view.on_changed = Some(Box::new(callback));
+            }
+        }
+        self
+    }
+
+}
+
+impl View for RadioButtons {
+    fn element(&self) -> Option<String> {
+        Some("radiobuttons".to_string())
+    }
+
+    fn event(&mut self, cx: &mut Context, event: &mut Event) {
+        if let Some(RadioButtonSelected(idx)) = event.message.downcast() {
+            if let Some(func) = self.on_changed.take() {
+                (func)(cx, *idx);
+                self.on_changed = Some(func);
+            }
+        }
+    }
+}
+
+struct RadioButton {
+    idx: usize,
+}
+
+impl View for RadioButton {
+    fn element(&self) -> Option<String> {
+        Some("radiobutton".to_owned())
+    }
+
+    fn event(&mut self, cx: &mut Context, event: &mut Event) {
+        if let Some(WindowEvent::MouseDown(btn)) = event.message.downcast() {
+            if *btn == MouseButton::Left {
+                cx.emit(RadioButtonSelected(self.idx));
+            }
+        }
+    }
+}
+
+impl RadioButton {
+    fn new<F>(cx: &mut Context, idx: usize, icon_text: String, builder: F) -> Handle<RadioButton>
+    where
+        F: 'static + FnOnce(&mut Context)
+    {
+        RadioButton {
+            idx
+        }.build2(cx, move |cx| {
+            let entity = cx.current;
+            entity.set_layout_type(cx, LayoutType::Row);
+            entity.set_height(cx, Stretch(0.0));
```

Comment:
The `build2` (will be renamed soon) method returns a `Handle` which you can set properties on directly. So instead of this you can do `build2(...).layout_type(LayoutType::Row).height(Stretch(0.0)`.

Also, I think you wan't `Stretch(1.0)` and not `Stretch(0.0)` unless I'm misunderstanding what you're trying to do.